### ISSUE TITLE
Formerly live videos fail to download: "failed to download [download] ... does not pass filter (live_status=?not_live); skipping"

### ISSUE THREAD

**Issue by holta** [iiab/calibre-web#188]
@nzola experienced many such errors.

Videos that were once live appear to be erroneously blocked by IIAB Calibre-Web.

Example of a video that should be downloading, but fails to download:
https://youtu.be/rbEsoe8F-l4
https://www.youtube.com/live/rbEsoe8F-l4 (possibly b/c YouTube prefers this "live" link???)

![IMG-20240615-WA0003~2](https://github.com/iiab/calibre-web/assets/2458907/2f413f77-2e5c-46ba-9abe-0ea740cea686)

@nzola mentioned:

> Downloaded these playlists [and] thumbnails without any problems: 
https://www.youtube.com/channel/UCX9j__vYOJu00iqBrCzecVw
https://www.youtube.com/playlist?list=PL1mP_vkqPB7EsIqqfwcGsg2rQNzoVy0mk

> But cannot download the following single videos:
https://www.youtube.com/watch?v=BK0XGf20l84
https://www.youtube.com/watch?v=VCM8tg_mGSw
https://www.youtube.com/watch?v=w8snrdaoTUs&t=2s
https://www.youtube.com/watch?v=5BO9nhtF0Cc
https://www.youtube.com/watch?v=rbEsoe8F-l4&t=7788s
https://www.youtube.com/watch?v=Drec4XAMJzI&t=6737s
https://www.youtube.com/watch?v=w8snrdaoTUs&t=7s

VM's iiab-diagnostics: https://dpaste.com/6H8F53GPQ

@deldesir: Any idea what's happening?

**Comment by deldesir** [comment]
2024-06-15 14:11:34 - [Debug] [https://www.youtube.com/watch?v=5BO9nhtF0Cc]: Unrecoverable error matched. [download] Analyse du 15 juin 2024: Controverse autour de la nomination du nouveau directeur de cabinet de FM-bM-^@M-& does not pass filter (live_status=?not_live);skipping ..

The error suggests the video is still a live video. On YouTube, the video is advertised as a finished live stream.

I bet the live_status column  in media table (xklb-metadata) says it's still a live.
For the moment, what I understand is the video need to pass the match filter per:

https://github.com/chapmanjacobd/library/blob/ffe4fc8cc97697356f3df01bd0e720d1e4176c91/xklb/createdb/tube_backend.py#L418

**Comment by holta** [comment]
Progress!

@nzola please also take note if **recently "live"** YouTube videos download OR do not download — in case these's a clear pattern as which succeed and which fail.

**Comment by nzola** [comment]
> Progress!
> 
> @nzola please also take note if _**recently "live"**_ YouTube videos download OR do not download — in case these's a clear pattern as to which succeed and which fail? 🙏

@holta @deldesir I understand that these videos were live when I tried to download them??? I will try downloading them again tomorrow, Sunday and give the report.

**Comment by nzola** [comment]
@deldesir I tried to download this playlist now:  https://www.youtube.com/@TOPCONGOFM/playlists It failed.
![image](https://github.com/iiab/calibre-web/assets/17815091/79987bab-c13b-44b3-a282-f34c992909f1)

Just remembered. I downloaded the same playlist yesterday, it failed 
![image](https://github.com/iiab/calibre-web/assets/17815091/a3458b88-967f-4b6a-9138-9fe5673cd6c4)

PUBLISHING TO URL... https://dpaste.com/H4SYGYT8U

**Comment by deldesir** [comment]
This is a playlist of playlists. I am trying to download it right now on my side (LRN2). Although only a subset of 100 videos will be attempted, the metadata fetch is going to take very long to gather metadata for all videos in every playlists found. I will report back on errors found. This may help me understanding the issues you have encountered.

**Comment by holta** [comment]
ASIDE: Of course YouTube videos that are truly "live" (happening at that moment) are not possible to download — until after the live event is complete. 🎤🎙️

**Comment by deldesir** [comment]
Downloading https://www.youtube.com/@TOPCONGOFM/playlists failed due to an unavailable video (
https://www.youtube.com/watch?app=desktop&v=n_iRE9al044). This needs a closer look. Thanks @Nzola for having reported on this.

**Comment by holta** [comment]
NON-URGENT:

"Tasks" view needs to use more clear language than  bureaucratic language like "unavailable video" eventually.

So regular teachers in all countries know what this really means 😇

Ideally with actionable suggestions (in those cases where that's realistic!)

**Comment by nzola** [comment]
> https://www.youtube.com/watch?v=Drec4XAMJzI&t=6737s

I tried again to download these 7 videos now , but still failed. I did not do any upgrade or new install.
PUBLISHING TO URL... https://dpaste.com/F6S6YX798

![image](https://github.com/iiab/calibre-web/assets/17815091/99fa8034-8402-48ed-99c2-a086eb365adf)

**Comment by holta** [comment]
> I tried again to download these 7 videos now , but still failed. I did not do any upgrade or new install.

Curiously the screenshot shows no error message / explanation at all, during this 2nd attempt.

(Definitely room for improvement, thanks @nzola.)

**Comment by deldesir** [comment]
The error message will be displayed now, but the videos will not download. Live videos (finished or not) are not downloadable with xklb by design.

**Comment by holta** [comment]
> Live videos (finished or not) are not downloadable with xklb by design.

By whose design?

FYI this design assumption seems extremely weak, given how videos labeled as "live" are actually used:

- Many who generate and post "live" material to YouTube (e.g. podcasters and similar) then leave these videos/podcasts/episodes/transmissions in YouTube's "live" section eternally.
- In short, as a very practical matter — the "live" label really means "this video was once live" in many/most cases.

**Comment by holta** [comment]
> The error message will be displayed now

@deldesir clarify which PR and code improved error reporting here?

Thanks, please if possible!

**Comment by deldesir** [comment]
> > The error message will be displayed now
> 
> @deldesir clarify which PR and code improved error reporting here?
> 
> Thanks, please if possible!

Per adjustments made in PR https://github.com/iiab/calibre-web/pull/194

**Comment by holta** [comment]
@deldesir please try to find a legit way to tell if a YouTube video is **_actually_** live or not.

_(Instead of the bogus information that we're currently using — that's as good as useless — given the fact that so many podcasters permanently leave their videos marked as "live" rather intentionally... !)_

**Comment by holta** [comment]
> @deldesir please try to find a legit way to tell if a YouTube video is actually live or not.

If this something like this can be upstreamed to become a part of xklb, even better!

✊

**Comment by holta** [comment]
@deldesir please use these URLs to make sure forward progress is steady in coming days — thank you to everyone working on this very common and very serious problem:

- https://www.youtube.com/@SugarlabsOrg-EN/streams a.k.a. their "Live" tab, containing formerly live videos
- https://github.com/iiab/calibre-web/issues/84#issuecomment-2183907512

**Comment by nzola** [comment]
> The error message will be displayed now, but the videos will not download. Live videos (finished or not) are not downloadable with xklb by design.

Ok. I understands. Thank you.

**Comment by holta** [comment]
@nzola @avni

Until we solve this serious problem properly...

@deldesir believes that an initial hack/workaround should confirm the path forward, using yt-dlp options like...

![Screenshot_20240622-162856](https://github.com/iiab/calibre-web/assets/2458907/af888d95-9b99-45a2-879a-88169bbf11ca)

https://github.com/yt-dlp/yt-dlp/blob/master/README.md

**Comment by nzola** [comment]
> @nzola @avni
> 
> Until we solve this serious problem properly...
> 
> @deldesir believes that an initial hack/workaround should confirm the path forward, using yt-dlp options like...
> 
> ![Screenshot_20240622-162856](https://private-user-images.githubusercontent.com/2458907/341996129-af888d95-9b99-45a2-879a-88169bbf11ca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkwODk1NjMsIm5iZiI6MTcxOTA4OTI2MywicGF0aCI6Ii8yNDU4OTA3LzM0MTk5NjEyOS1hZjg4OGQ5NS05Yjk5LTQ1YTItODc5YS04ODE2OWJiZjExY2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjJUMjA0NzQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Yzk3NDJmNzI1MDYzYjg0MTg3YWRlYTAwNjliZGEzNDk2M2QwN2Y3ZGU2ODlhZTE1OTQ1NWZjMDUzYTQwNmRhOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.lAabkCXAnflYHyjyDLyXidwEOUGN2LjsrEQ73eiK_8M)
> 
> https://github.com/yt-dlp/yt-dlp/blob/master/README.md

Ok @holta

**Comment by nzola** [comment]
> @nzola experienced many such errors.
> 
> Videos that were once live appear to be erroneously blocked by IIAB Calibre-Web.
> 
> Example of a video that should be downloading, but fails to download:
> 
> * https://youtu.be/rbEsoe8F-l4
> * https://www.youtube.com/live/rbEsoe8F-l4 (false positive possibly b/c YouTube prefers this "live" link???)
> 
> ![IMG-20240615-WA0003~2](https://private-user-images.githubusercontent.com/2458907/340044963-2f413f77-2e5c-46ba-9abe-0ea740cea686.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkwOTEzMjQsIm5iZiI6MTcxOTA5MTAyNCwicGF0aCI6Ii8yNDU4OTA3LzM0MDA0NDk2My0yZjQxM2Y3Ny0yZTVjLTQ2YmEtOWFiZS0wZWE3NDBjZWE2ODYuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjJUMjExNzA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjVlYTFlOTk0ZTFjOTNkNWJlZTdiMGZmY2MzYmM3NDIzMjA3NTJhMTQ2MGRlMzJjMDhiMDI0ZmRlODQxMmM3NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.wXHPnJIGdxrcrgC_95XLRKQB3j_DRS93Ukw68ypd4no)
> 
> @nzola mentioned:
> 
> > Downloaded these playlists [and] thumbnails without any problems:
> > https://www.youtube.com/channel/UCX9j__vYOJu00iqBrCzecVw
> > https://www.youtube.com/playlist?list=PL1mP_vkqPB7EsIqqfwcGsg2rQNzoVy0mk
> 
> > But cannot download the following single videos:
> > https://www.youtube.com/watch?v=BK0XGf20l84
> > https://www.youtube.com/watch?v=VCM8tg_mGSw
> > https://www.youtube.com/watch?v=w8snrdaoTUs&t=2s
> > https://www.youtube.com/watch?v=5BO9nhtF0Cc
> > https://www.youtube.com/watch?v=rbEsoe8F-l4&t=7788s
> > https://www.youtube.com/watch?v=Drec4XAMJzI&t=6737s
> > https://www.youtube.com/watch?v=w8snrdaoTUs&t=7s
> 
> VM's iiab-diagnostics: https://dpaste.com/6H8F53GPQ
> 
> @deldesir: Any idea what's happening?

JFYI: I downloaded all these videos on CMD with yt-dlp.

**Comment by holta** [comment]
@deldesir

- @nzola suggests the challenge is not downloading "formerly live" YouTube videos...

- But rather, the challenge appears to be (programmatically, reliably) identifying "formerly live" YouTube videos...

_Can you confirm?_

**Comment by avni** [comment]
iiab-diagnostics: https://dpaste.com/7XBX3XZE2

I can replicate the error, and see, "failed to download [download] ... does not pass filter (live_status=?not_live); skipping" when downloading:

1. https://www.youtube.com/watch?v=rbEsoe8F-l4
2. https://www.youtube.com/live/rbEsoe8F-l4

>@deldesir believes that an initial hack/workaround should confirm the path forward, using yt-dlp options like...

It's not clear to me what the workaround is or what I need to do to test? Is the ask to use the command line and testing the download with yt-dlp? 

![Image](https://github.com/user-attachments/assets/4f696b51-4825-4a73-b78b-b60267db88f4)

**Comment by holta** [comment]
@deldesir should outline a suggested path forward within 24h.

- As mentioned above, downloading via yt-dlp seems to work fine, no options/flags required.

- The real issue seems to be automatically distinguishing between "truly live" YouTube videos and "formerly live" YouTube videos.

- And then presumably pushing functionality upstream, e.g. into xklb?

**Comment by avni** [comment]
>As mentioned above, downloading "formerly live" videos via yt-dlp seems to work fine, no options/flags required.

Here's the output from my terminal as confirmation. I don't know how to view the webm files, but presuming they succeeded based on the output.

> ubuntu@box: $ sudo yt-dlp https://www.youtube.com/watch?v=rbEsoe8F-l4
> [youtube] Extracting URL: https://www.youtube.com/watch?v=rbEsoe8F-l4
> [youtube] rbEsoe8F-l4: Downloading webpage
> [youtube] rbEsoe8F-l4: Downloading ios player API JSON
> [youtube] rbEsoe8F-l4: Downloading m3u8 information
> [info] rbEsoe8F-l4: Downloading 1 format(s): 247+251
> [download] Destination: LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].f247.webm
> [download] 100% of  541.37MiB in 00:01:31 at 5.90MiB/s
> [download] Destination: LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].f251.webm
> [download] 100% of  126.31MiB in 00:00:47 at 2.67MiB/s
> [Merger] Merging formats into "LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].webm"
> Deleting original file LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].f251.webm (pass -k to keep)
> Deleting original file LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].f247.webm (pass -k to keep)
> ubuntu@box:~$ ls
> 'LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].webm'


> ubuntu@box: $ sudo yt-dlp https://www.youtube.com/live/rbEsoe8F-l4
> [youtube] Extracting URL: https://www.youtube.com/live/rbEsoe8F-l4
> [youtube] rbEsoe8F-l4: Downloading webpage
> [youtube] rbEsoe8F-l4: Downloading ios player API JSON
> [youtube] rbEsoe8F-l4: Downloading m3u8 information
> [info] rbEsoe8F-l4: Downloading 1 format(s): 247+251
> [download] LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].webm has already been downloaded
> ubuntu@box: $ rm 'LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].webm' 
> rm: remove write-protected regular file 'LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].webm'? yes
> ubuntu@box:~$ sudo yt-dlp https://www.youtube.com/live/rbEsoe8F-l4
> [youtube] Extracting URL: https://www.youtube.com/live/rbEsoe8F-l4
> [youtube] rbEsoe8F-l4: Downloading webpage
> [youtube] rbEsoe8F-l4: Downloading ios player API JSON
> [youtube] rbEsoe8F-l4: Downloading m3u8 information
> [info] rbEsoe8F-l4: Downloading 1 format(s): 247+251
> [download] Destination: LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].f247.webm
> [download] 100% of  541.37MiB in 00:00:19 at 28.26MiB/s
> [download] Destination: LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].f251.webm
> [download] 100% of  126.31MiB in 00:00:19 at 6.53MiB/s
> [Merger] Merging formats into "LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].webm"
> Deleting original file LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].f247.webm (pass -k to keep)
> Deleting original file LE DEBAT 24 MAI 2024 [rbEsoe8F-l4].f251.webm (pass -k to keep)

**Comment by holta** [comment]
Progress brewing!

- PR chapmanjacobd/library#41
- PR #199

**Comment by deldesir** [comment]
> 1. Please confirm/clarify the question @nzola [surfaced](https://github.com/iiab/calibre-web/issues/188#issuecomment-2184231321) 3 days ago — isn't the real problem here programmatically identifying "actually live" versus "formerly live" YouTube videos?

You mean the `live_status` of the video, whether it's `live` or `was_live`? Absolutely not. The real problem is because the `live_status` column in xklb-metadata.db doesn't have the value `not_live`. The match_filter from xklb in https://github.com/chapmanjacobd/library/blob/e9975b07bca5b481aeed9398d0bc0adb3b9b25c8/xklb/createdb/tube_backend.py#L418 forces yt-dlp to reject/skip the download because the video doesn't have this `not_live` status/criterion.

> 2. How will downloading work with "actually live" YouTube videos (i.e. whose recording is not yet complete) — e.g. as outlined in the 12-hour tennis match question at [Add --live option chapmanjacobd/library#41 (comment)](https://github.com/chapmanjacobd/library/pull/41#issuecomment-2186701059) ?

Once chapmanjacobd/library#41, if ever it is approved, live videos will download on IIAB Calibre-Web. Additional testing will be needed to ensure performance is not overly affected by these generally huge recordings.

**Comment by holta** [comment]
@deldesir let me ask my question at https://github.com/chapmanjacobd/library/pull/41#issuecomment-2186701059 more directly:

- People who click on "Download to IIAB" need to know whether a YouTube video is "actually live" versus "formerly live" — otherwise they could (apparently, if I'm understanding) be stuck waiting many hours until the "actually live" video is finally downloaded in full.
- So, how can we programmatically distinguish between "actually live" and "formerly live" YouTube videos, to help these people avoid getting stuck waiting for perhaps up-to-12 hours?

**Comment by holta** [comment]
CAVEAT: I concede that some people who click on "Download to IIAB" don't mind extremely slow downloads of "actually live" videos.

- One concrete example is radio show podcasts, that are almost always complete within about a single hour.
- So the person clicking "Download to IIAB" to download that radio show or podcast, will often be very happy to wait an hour-or-so for IIAB Calibre-Web to slowly download that episode (that's still is happening, it's literally live).
- Conversely, some others might realize that they're downloading a 3-hour sports match — so back to my question: _should we (and can we?) notify users within "Tasks" view that they've begun downloading an "actually live" YouTube video — that might possibly take hours-and-hours to finish downloading?_

**Comment by holta** [comment]
> Additional testing will be needed to ensure performance is not overly affected by these generally huge recordings.

Yes that issue of overweight videos is very important.  And very well known since 2023[*].  But please let's declare that separate question (or overly long, overlie bandwidth-heavy and disk-heavy videos) to be off-topic — at least in this particular context here: 😉

- #79

[*] ASIDE: Hopefully to be solved in a few short months, or possibly much earlier!

**Comment by holta** [comment]
ALSO: Pre-announced / Pre-scheduled / Upcoming videos ("not yet live" !) are a whole other category of YouTube URL's... 

...that I didn't realize also matter!

_(Presumably these too are commonly categorized as "live" videos, even though no video whatsoever exists yet!?)_

▶️ Certainly we need some kind of intelligent user-facing warning or messaging — in "Tasks" view or similar — to warn teachers / parents when they're trying to "Download to IIAB" a video that doesn't yet exist! ⏳

https://youtube.com/channel/UC4R8DWoMoI7CAwX8_LjQHig

**Comment by holta** [comment]
3 kinds of "ostensibly live" YouTube videos tested here:

- https://github.com/iiab/calibre-web/pull/199#issuecomment-2196754732

With PR #199 now merged, a follow-up PR is now needed to clean up, as described here:

- #202

**Comment by nzola** [comment]
> @nzola experienced many such errors.
> 
> Videos that were once live appear to be erroneously blocked by IIAB Calibre-Web.
> 
> Example of a video that should be downloading, but fails to download:
> 
> * https://youtu.be/rbEsoe8F-l4
> * https://www.youtube.com/live/rbEsoe8F-l4 (false positive possibly b/c YouTube prefers this "live" link???)
> 
> ![IMG-20240615-WA0003~2](https://private-user-images.githubusercontent.com/2458907/340044963-2f413f77-2e5c-46ba-9abe-0ea740cea686.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk3MjM3NDgsIm5iZiI6MTcxOTcyMzQ0OCwicGF0aCI6Ii8yNDU4OTA3LzM0MDA0NDk2My0yZjQxM2Y3Ny0yZTVjLTQ2YmEtOWFiZS0wZWE3NDBjZWE2ODYuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MzBUMDQ1NzI4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjllMTBiM2M2MDY5ZTQwYzg3N2RmOGJhMDJmMGZhYzVhMDMxYmE1MWFlMmVkOGJkMTk0ODZhNDA3ODI5NWY5YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NRjra7oPqXpVxxVcgeHMTHk5IPHt-Y8DHSmC4OrACi4)
> 
> @nzola mentioned:
> 
> > Downloaded these playlists [and] thumbnails without any problems:
> > https://www.youtube.com/channel/UCX9j__vYOJu00iqBrCzecVw
> > https://www.youtube.com/playlist?list=PL1mP_vkqPB7EsIqqfwcGsg2rQNzoVy0mk
> 
> > But cannot download the following single videos:
> > https://www.youtube.com/watch?v=BK0XGf20l84
> > https://www.youtube.com/watch?v=VCM8tg_mGSw
> > https://www.youtube.com/watch?v=w8snrdaoTUs&t=2s
> > https://www.youtube.com/watch?v=5BO9nhtF0Cc
> > https://www.youtube.com/watch?v=rbEsoe8F-l4&t=7788s
> > https://www.youtube.com/watch?v=Drec4XAMJzI&t=6737s
> > https://www.youtube.com/watch?v=w8snrdaoTUs&t=7s
> 
> VM's iiab-diagnostics: https://dpaste.com/6H8F53GPQ
> 
> @deldesir: Any idea what's happening?

The playlist downloaded from 1 to 100% then it failed
PUBLISHING TO URL... https://dpaste.com/4HVB2SFVK

![image](https://github.com/iiab/calibre-web/assets/17815091/df47ed17-a083-477b-9d3b-0c990005934c)

**Comment by holta** [comment]
> playlist https://www.youtube.com/@TOPCONGOFM/playlists downloaded from 1 to 100% then it failed

@nzola the above is not a playlist.  It is a _**list of playlists**._  Can you try again with one of its individual playlists?  For example, maybe start with their 8-video playlist https://youtube.com/playlist?list=PLi6Z1Wvj99SKPHMiqLkREMQrMyZy1YTPE ?

@deldesir can you please clean up the error message so that @nzola and others have a crystal clear error message?  (We can open this message in a new ticket if that makes things easier!)

Conversely: If a _**list of playlists**_ can easily in future be downloaded into a single bookshelf, much like we do with a YouTube channel (a YouTube channel is basically also a list of playlists!) then we can consider that functionality, if it's genuinely needed?

**Comment by holta** [comment]
@nzola and AFTER you've completed testing of individual playlist(s):

Try the much more ambitious experiment of downloading the entire channel — e.g. try the "Download to IIAB" button with URLs like:

- https://youtube.com/@TOPCONGOFM
- https://youtube.com/c/TOPCONGOFM

**Comment by nzola** [comment]
Downloaded this playlist:https://www.youtube.com/playlist?list=PLi6Z1Wvj99SLJRvpu2S6CPRHvECynHS0y
(99 videos) with **pi400 Calibre-Web**. Successfully completed after 8 hours.
PUBLISHING TO URL... https://dpaste.com/3LGEYAQLE
![image](https://github.com/iiab/calibre-web/assets/17815091/d69e5200-c351-437e-8d1b-1249cd7ed6eb)

This error happened during the download, but the process continued until all 99 videos were completed download.

![image](https://github.com/iiab/calibre-web/assets/17815091/71df5682-eca7-45aa-8c83-d74c2d8eb6cc)
![image](https://github.com/iiab/calibre-web/assets/17815091/d68f14b8-e0da-4ed9-b84d-2fb684261832)

**Comment by nzola** [comment]
@deldesir @holta **pi4 Calibre-Web** also downloaded this playlist: https://www.youtube.com/playlist?list=PLi6Z1Wvj99SKvAGNRiDmjhgQEZ8pU-4Y2 very smoothly without any problems
PUBLISHING TO URL... https://dpaste.com/H37ZJTM7R

**Comment by nzola** [comment]
> @nzola and AFTER you've completed testing of individual playlist(s):
> 
> Try the much more ambitious experiment of downloading the entire channel — e.g. try the "Download to IIAB" button with URLs like:
> 
> * https://youtube.com/@TOPCONGOFM
> * https://youtube.com/c/TOPCONGOFM

@deldesir @holta I tried downloading the 2 above video links on both **pi4 and multipass Calibre-Web**, but they are still stuck on STARTED. 
pi4 Calibre-Web results: PUBLISHING TO URL... https://dpaste.com/H37ZJTM7R

![image](https://github.com/iiab/calibre-web/assets/17815091/8b785de9-603f-43a2-ad5e-ddbb0100b677)

multipass Calibre-Web results:

**Comment by holta** [comment]
> Downloaded this playlist:https://www.youtube.com/playlist?list=PLi6Z1Wvj99SLJRvpu2S6CPRHvECynHS0y (99 videos) with **pi400 Calibre-Web**. Successfully completed after 8 hours. PUBLISHING TO URL... https://dpaste.com/3LGEYAQLE
> [SCREENSHOT OF SUCCESSES] 
> This error happened during the download, but the process continued until all 99 videos were completed download.
> [SCREENSHOTS OF FAILURE]

- @nzola great news that 99 out of 100 (of 230 total) downloaded!
- Can you test "Download to IIAB" of the single failed video https://youtu.be/Zm-ROp6EMZo all on its own?  (To see if that failure is repeatable or not!)

**Comment by holta** [comment]
@deldesir were channel downloads like this working in the past?

> > * https://youtube.com/@TOPCONGOFM
> > * https://youtube.com/c/TOPCONGOFM
> 
> @deldesir @holta I tried downloading the 2 above video links on both **pi4 and multipass Calibre-Web**, but they are still stuck on STARTED. pi4 Calibre-Web results: PUBLISHING TO URL... https://dpaste.com/H37ZJTM7R
> [SCREENSHOT OF "Metadata Fetch" THAT GETS STUCK "Waiting" ON RPI 4]
>
> multipass Calibre-Web results: PUBLISHING TO URL... https://dpaste.com/GFTD2QH9D
> [SCREENSHOT OF "Metadata Fetch" THAT GETS STUCK "Waiting" ON 24.04 VM]

**Comment by holta** [comment]
> @deldesir were channel downloads like this working in the past?

@EMG70 wonders if part of the reason is...

> Download stuck on "Waiting "I thought this has been normal behaviour when one tries to download another video before the first one has started or finished gathering meta data.

> The long wait happens mostly on channels or extremely long playlists such that any attempt to download a second url will give the "waiting" status

(@deldesir can you help clarify?!)

**Comment by deldesir** [comment]
> This error happened during the download, but the process continued until all 99 videos were completed download.

Very ugly. Likely a network error.

**Comment by deldesir** [comment]
> @deldesir were channel downloads like this working in the past?
 * https://youtube.com/@TOPCONGOFM
 * https://youtube.com/c/TOPCONGOFM

We did tests with channels in the past but not big as https://youtube.com/@TOPCONGOFM. This channel has 4.1k videos. Even though only 100 videos will be downloaded, the first task will take a long time processing metadata for all those 4.1k videos.

**Comment by deldesir** [comment]
The long wait is justified as I could see all playlists being indexed running `tail -f /var/log/xklb.log`. Test done in lrn2. I had to reboot the machine. I guess I'll have to move the database aside too because a lot of residual videos will be caught in subsquent downloads.

**Comment by holta** [comment]
@EMG70 further explains https://github.com/iiab/calibre-web/issues/188#issuecomment-2200213122

> Just to be sure we are not investigating normal behaviour I initiated  a 3hr video download, before it completed gathering meta I started a 16min video download. This resulted in second video going to waiting mode and only started after first video was completed as what's happening on @nzola's channel downloads.

![image](https://github.com/iiab/calibre-web/assets/2458907/59030f49-d734-4c51-a4a7-e66a97d1a8ef)

![image](https://github.com/iiab/calibre-web/assets/2458907/7bab1cdb-5df2-4cc4-be09-951b5355a7cd)

**Comment by deldesir** [comment]
@EMG70 explained it well. Videos are downloaded sequentially. I was investigating the channel having a "started" message with a progress stuck at 0%.

**Comment by nzola** [comment]
I tried to download this video link as requested by @holta on multipass iiab, pi4 iiab and pi400 iiab.
Here are the results:

multipass iiab: PUBLISHING TO URL... https://dpaste.com/3R8RXUUZ2
![image](https://github.com/iiab/calibre-web/assets/17815091/921258f9-36d4-4cf2-9f39-5b9f73649a54)

pi4: PUBLISHING TO URL... https://dpaste.com/2VXWWLBZB
![image](https://github.com/iiab/calibre-web/assets/17815091/8432fb58-0664-4152-a245-1ef4be7be46e)

pi400: PUBLISHING TO URL... https://dpaste.com/8Z4TA84FH
![image](https://github.com/iiab/calibre-web/assets/17815091/2e6d254c-3b35-4a10-823e-8dce7f58a4e6)
